lifecycle/src/lifecycle_talker.cpp

```diff
@@ -195,7 +195,7 @@ class LifecycleTalker : public rclcpp_lifecycle::LifecycleNode
     return lifecycle_msgs::msg::Transition::TRANSITION_CALLBACK_SUCCESS;
   }
 
-  /// Transition callback for state deactivating
+  /// Transition callback for state cleanup
```

Comment:
should this be "cleaningup" instead of "cleanup" ?

Reply:
I think cleanup reads better myself.

Reply:
I agree with the fact that it reads better. I figured we wanted to match the name of the transition state defined in the [design document](http://design.ros2.org/articles/node_lifecycle.html) and [in the same docblock below](https://github.com/ros2/demos/pull/221/files#diff-a333d83f31412712a00d05e83145ebb0R201) but am fine with both

Reply:
Ok, yeah for that it does make sense. It is the "cleaning up" state not the "cleanup of state".

Reply:
So what's the conclusion? The callback name and the state name should probably match, and ideally they would also match the design document.

Reply:
:+1: yes the conclusion is to use "cleaningup" to match the design doc and the rest of the code

Reply:
OK, done. :+1: 

I think the callback name in the demo and the API should also be changed to `on_cleaningup`, but that's a separate issue.

Reply:
>  think the callback name in the demo and the API should also be changed to on_cleaningup, but that's a separate issue.

I'm going to merge this now but feel free to open a PR to address those.

Reply:
> I think the callback name in the demo and the API should also be changed to on_cleaningup, but that's a separate issue.

Actually based on the article, I'd say the callback name is right, no? It's this:

![capture](https://user-images.githubusercontent.com/100427/34832941-185bb3b4-f6a3-11e7-8ec2-074bdea196cb.PNG)

State is "CleaningUp" but the transition is called "onCleanup".

Reply:
:+1:  reading the design doc I have the same reading as @wjwwood

Reply:
I'm happy with that.